### PR TITLE
EXOtrigger MET & HT prompt monitoring 92X

### PR DESCRIPTION
**Pull request by sudhaahuja**
Update with the METmonitoring and HTmonitoring codes for the offline DQM monitoring (EXO trigger paths added)

### CODE REVIEW

**Comment by arunhep** [comment]
@sudhaahuja i think it needs to be rebased.

**Comment by arunhep** [comment]
-1

**Comment by fwyzard** [comment]
please rebase

**Comment by mtosi** [comment]
i will check the list of paths at CERN
(w/ the phone is not that straightforward)

i've just had a worry/doubt
you are using only one jets collection,
but do we have case in which one make use of both the AK4 and AK8 collections ?
for instances, one used for the offline selections and the other for the HT computation ?
or maybe both for the offline selections ?

can you comment on this, please ?

thanks !!

**Comment by rappoccio** [comment]
Hi, 

Also, I still do not see this file:

from DQMOffline.Trigger.htMonitoring_cfi import htMonitoring


It is not in the release, did you forget to commit it?

**Comment by mtosi** [comment]
are you refering to htMonitoring ?

it is automatically created by the FillDescription

**Comment by mtosi** [comment]
I think you need to rebase ....

**Comment by mtosi** [comment]
I still do not see the following paths
what am I missing ?

HLT_PFMETTypeOne140_PFMHT140_IDTight_v1
HLT_PFMETNoMu130_PFMHTNoMu130_IDTight_v9
HLT_PFMETNoMu140_PFMHTNoMu140_IDTight_v9
HLT_MonoCentralPFJet80_PFMETNoMu130_PFMHTNoMu130_IDTight_v9
HLT_MonoCentralPFJet80_PFMETNoMu140_PFMHTNoMu140_IDTight_v9
HLT_PFHT890_v7
HLT_PFHT780_v7
HLT_PFHT680_v7
HLT_PFHT590_v7
HLT_PFHT510_v7
HLT_PFHT430_v7
HLT_PFHT370_v7
HLT_PFHT250_v7
HLT_PFHT180_v7

**Comment by rappoccio** [comment]
@mtosi OK, so if the code gets correctly set, we can just clone it for the B2G setup? Thanks.

**Comment by mtosi** [comment]
I would say so
you will probably need to update / adjust / overwrite the relevant
parameters
(above all the genericTriggerEventFlag settings)
  mia

**Comment by rappoccio** [comment]
@mtosi yes, we did that. OK, we're basically ready to go, just waiting for the rebase here and we can submit the B2G triggers. 

@alefisico we're just about done.

**Comment by mtosi** [comment]
thanks !!!
  mia

**Comment by alefisico** [comment]
@rappoccio great Sal, thanks

**Comment by sudhaahuja** [comment]
@mtosi - to answer your question - In EXO, only Alejandro's paths are using AK8 jets, so we can look into this with him once he requests the integration of his paths.

**Comment by arunhep** [comment]
this PR needs a rebase. @sudhaahuja can you please act on it?

**Comment by mtosi** [comment]
@sudhaahuja
can you please update this PR ?

**Comment by alefisico** [comment]
sorry @sudhaahuja which implementation are you talking about? the substructure triggers have already a JIRA ticket with the implementation in the menu: https://its.cern.ch/jira/browse/CMSHLT-1262

**Comment by mtosi** [comment]
@dmitrijus can we start the integration test, please ?

**Comment by fwyzard** [comment]
tracked by #19142

**Comment by fwyzard** [comment]
tracked by #19142

**Comment by fwyzard** [comment]
please test

**Comment by mtosi** [comment]
what about 
HLT_PFHT890_v*
HLT_PFHT780_v*
HLT_PFHT680_v*
HLT_PFHT590_v*
HLT_PFHT510_v*
HLT_PFHT430_v*
HLT_PFHT370_v*
HLT_PFHT250_v*
HLT_PFHT180_v*
?!?!?!?

**Comment by mtosi** [comment]
please, rebase

**Comment by sudhaahuja** [comment]
@mtosi - rebase done

**Comment by fwyzard** [comment]
@cmsbuild please test

**Comment by sudhaahuja** [comment]
Also, @mtosi - the PFHT paths you are mentioning were already included in the branch sometime ago  -https://github.com/sudhaahuja/cmssw/blob/EXOTrigger_promptMonitoring92X/DQMOffline/Trigger/python/HTMonitor_cff.py#L118-#L170

**Comment by dmitrijus** [comment]
how is this related to https://github.com/cms-sw/cmssw/pull/18971 ?

**Comment by gomber** [comment]
@dmitrijus  it has nothing to do with #18971 , i think you got confused with the same branch name. But @sudhaahuja is only referring to her PR.

**Comment by sudhaahuja** [comment]
There was a conflict in the file - ExoticaMonitoring_Client_cff.py earlier and resolving it directly on the PR resulted in few lines being deleted from the file, which led to the errors this time. Sorry for that.

**Comment by davidlange6** [comment]
@sudhaahuja @gomber - if this and #18971 are not the same thing, please change their titles to be more descriptive of the changes. Thanks

**Comment by dmitrijus** [comment]
+1

**Comment by mtosi** [comment]
what about 

HLT_PFHT890_v*
HLT_PFHT780_v*
HLT_PFHT680_v*
HLT_PFHT590_v*
HLT_PFHT510_v*
HLT_PFHT430_v*
HLT_PFHT370_v*
HLT_PFHT250_v*
HLT_PFHT180_v*

**Comment by sudhaahuja** [comment]
@mtosi - you can find those paths in the HT monitor here - https://github.com/sudhaahuja/cmssw/blob/EXOTrigger_promptMonitoring92X/DQMOffline/Trigger/python/HTMonitor_cff.py#L118-#L170

**Comment by mtosi** [comment]
thanks, I've just re-checked in this PR, and I saw them ... sorry for not having spotted them before :(

can you please
1. fix the label of some paths
2. check the binning you use for the different HT threshold ?
thanks

File: DQMOffline/Trigger/python/HTMonitor_cff.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+# HLT_MonoCentralPFJet80_PFMETNoMu120_PFMHTNoMu120_IDTight
+MonoCentralPFJet80_PFMETNoMu120_PFMHTNoMu120_HTmonitoring = hltHTmonitoring.clone()
+MonoCentralPFJet80_PFMETNoMu120_PFMHTNoMu120_HTmonitoring.FolderName = cms.string('HLT/HT/MonoCentralPFJet80_PFMETNoMu120/')
+MonoCentralPFJet80_PFMETNoMu120_PFMHTNoMu120_HTmonitoring.numGenericTriggerEventPSet.hltPaths = cms.vstring("HLT_MonoCentralPFJet80_PFMETNoMu120_PFMHTNoMu120_IDTight_v*")
+MonoCentralPFJet80_PFMETNoMu120_PFMHTNoMu120_HTmonitoring.jetSelection_HT   = cms.string("pt > 30 && eta < 2.5")
+
+# HLT_MonoCentralPFJet80_PFMETNoMu130_PFMHTNoMu130_IDTight
+MonoCentralPFJet80_PFMETNoMu130_PFMHTNoMu130_HTmonitoring = hltHTmonitoring.clone()
+MonoCentralPFJet80_PFMETNoMu130_PFMHTNoMu130_HTmonitoring.FolderName = cms.string('HLT/HT/MonoCentralPFJet80_PFMETNoMu130/')
+MonoCentralPFJet80_PFMETNoMu130_PFMHTNoMu130_HTmonitoring.numGenericTriggerEventPSet.hltPaths = cms.vstring("MonoCentralPFJet80_PFMETNoMu130_PFMHTNoMu130_IDTight_v*")
```
Comment (fwyzard) [inline DQMOffline/Trigger/python/HTMonitor_cff.py]: please fix the path name to `HLT_MonoCentralPFJet80_PFMETNoMu130_PFMHTNoMu130_IDTight_v*`

File: DQMOffline/Trigger/python/HTMonitor_cff.py [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+MonoCentralPFJet80_PFMETNoMu130_PFMHTNoMu130_HTmonitoring.jetSelection      = cms.string("pt[0] > 100 && eta[0] < 2.5 && neutralHadronEnergyFraction[0] < 0.8 && chargedHadronEnergyFraction[0] > 0.1")
+MonoCentralPFJet80_PFMETNoMu130_PFMHTNoMu130_HTmonitoring.muoSelection      = cms.string("isTightMuon()")
+MonoCentralPFJet80_PFMETNoMu130_PFMHTNoMu130_HTmonitoring.jetSelection_HT   = cms.string("pt > 30 && eta < 2.5")
+
+# HLT_MonoCentralPFJet80_PFMETNoMu140_PFMHTNoMu140_IDTight
+MonoCentralPFJet80_PFMETNoMu140_PFMHTNoMu140_HTmonitoring = hltHTmonitoring.clone()
+MonoCentralPFJet80_PFMETNoMu140_PFMHTNoMu140_HTmonitoring.FolderName = cms.string('HLT/HT/MonoCentralPFJet80_PFMETNoMu140/')
+MonoCentralPFJet80_PFMETNoMu140_PFMHTNoMu140_HTmonitoring.numGenericTriggerEventPSet.hltPaths = cms.vstring("_v*")
```
Comment (fwyzard) [inline DQMOffline/Trigger/python/HTMonitor_cff.py]: please fix the path name to `HLT_MonoCentralPFJet80_PFMETNoMu140_PFMHTNoMu140_IDTight_v*`

**Review by fwyzard — CHANGES_REQUESTED** [review]

**Comment by fwyzard** [comment]
do you have the monitoring for
```
HLT_PFMETNoMu110_PFMHTNoMu110_IDTight_v*
HLT_PFMETNoMu130_PFMHTNoMu130_IDTight_v*
HLT_PFMETNoMu140_PFMHTNoMu140_IDTight_v*
```
?

**Comment by rappoccio** [comment]
@sudhaahuja we're really running out of time here and we have B2G triggers that are also dependent upon this. What's the status? Do you need help with this?

**Comment by sudhaahuja** [comment]
@rappoccio, others - sorry for the delay. I am having night shifts these days... I am working on it now & will commit the changes in a short while...

**Comment by sudhaahuja** [comment]
@rappoccio, others - sorry for the delay. I am having night shifts these days... I am working on it now & will commit the changes in a short while...

**Comment by rappoccio** [comment]
Thanks, if you need help, let us know. We can help out if needed.

**Comment by sudhaahuja** [comment]
@mtosi @rappoccio @UBParker - the PR was updated with the changes/corrections suggested. I also added the B2G Monitor files provided by Ashley from this PR #19214 . @fwyzard - these triggers HLT_PFMETNoMu110_PFMHTNoMu110_IDTight_v*
HLT_PFMETNoMu130_PFMHTNoMu130_IDTight_v*
HLT_PFMETNoMu140_PFMHTNoMu140_IDTight_v*  , can be found inside MET monitor here - https://github.com/sudhaahuja/cmssw/blob/EXOTrigger_promptMonitoring92X/DQMOffline/Trigger/python/METMonitor_cff.py#L68-#L94

**Comment by Martin-Grunewald** [comment]
-1
rebase properly

**Comment by rappoccio** [comment]
@sudhaahuja thank you. Let us know if you need help with anything.

**Comment by rappoccio** [comment]
@Martin-Grunewald I have this correctly rebased on a branch in my local repo. Is there a way to update THIS PR with that code, or should I make a new PR?

**Comment by Martin-Grunewald** [comment]
Hi,

It looks to me that the list of commits includes commits from other PRs which should not be here, eg, https://github.com/cms-sw/cmssw/pull/18978/commits/815ec6fad1d3a87d5fe47e2cb0729caf63404ebd

That's why I think a rebase is needed (I assume you added these commits by committing to this branch from a later IB release area than your original, thus adding those spurious commits)

**Comment by rappoccio** [comment]
@Martin-Grunewald Yes, that is what @sudhaahuja did, I suspect. However, she is now traveling. I have rebased these correctly on my fork. Should I make a new PR or is there a way to replace the commits in this one?

**Comment by rappoccio** [comment]
I'm not sure how to proceed, so I opened the other PR: #19293. Should this one be closed?

**Comment by Martin-Grunewald** [comment]
yes...

**Pull request closed without merging**